app/features/settings/profile/editable-username.tsx

```diff
@@ -56,11 +57,20 @@ export default function EditableUsername() {
       await updateUsername(data.username);
       startAnimation();
       setIsEditing(false);
-    } catch {
-      toast({
-        title: 'Error',
-        description: 'Failed to update username',
-      });
+    } catch (error) {
+      if (error instanceof UniqueConstraintError) {
+        toast({
+          title: 'Error',
+          description: 'Username already taken',
```

Comment:
I would maybe just say `Username not available` instead because in the future we will probably also have some blacklist for usernames so we can use the same message for that and username being taken by another user